kms_utils.py

```diff
@@ -52,36 +52,42 @@ def check_token_needs_renewal(force):
   renewal = True
   jsonInfo = get_token_info()
   creationTime = jsonInfo['data']['creation_time']
+  
+  #Convert time as given from Vault to epoch time
+  expire_time_vault = jsonInfo['data']['expire_time']
+  expire_time_string = expire_time_vault.split('T')[0] + ' ' + expire_time_vault.split('T')[1].split('.')[0]
```

Comment:
This is unnecessary, you could parse directly from expire_time_vault with this format '%Y-%m-%dT%H:%M:%S'